### PR TITLE
Added Analytics and Posthog packages

### PR DESCRIPTION
**Pull request by dlymonk**
## Checklist before requesting a review

- [x] I have updated the unit tests based on the changes I made
- [x] I have updated the docs (TSDoc / README / global doc) to reflect my changes
- [x] I have performed self-QA of my feature by testing the apps and packages and made sure that :
  - No regression or new bug has occurred
  - The acceptance criteria listed in the ticket are met
  - **Self-QA was made on both desktop and mobile**

### CODE REVIEW

File: documentation/docs/packages/camera-web.md [hunk 1]
```diff
@@ -1,13 +1,15 @@
 ---
-sidebar_position: 1
+sidebar_position: 2
 ---
 
 # camera-web
+
```
Comment (souyahia-monk) [inline documentation/docs/packages/camera-web.md:6]: Why did you add these blank lines under every title in the different markdown documents? My guess is that it's an auto-correct thing made by your IDE? Remove all of these blank lines,, there are a lot in the PR

File: packages/analytics/src/adapters/adapter.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+/**
+ * Primitive JavaScript type used in analytics tags.
+ */
+export type Primitive = number | string | boolean | bigint | symbol | null | undefined;
```
Comment (souyahia-monk) [inline packages/analytics/src/adapters/adapter.ts:4]: This type already exist in the Monitoring package. When you have two packages that use the same type, you must extract this type and declare it in the types package

File: packages/analytics/src/adapters/adapter.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+/**
+ * Interface describing the requirements for a Analytics Adapter. An adapter is a tool (such as Posthog etc.) that
+ * allow the application to track event for product analysis.
```
Comment (souyahia-monk) [inline packages/analytics/src/adapters/adapter.ts:8]: ...for a**n** Analytics Adapter...

...that allow**s**...

File: packages/analytics/src/adapters/adapter.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ */
+export interface AnalyticsAdapter {
+  /**
+   * Set the ID of the user currently using the application. Ignore this if you have no user system.
+   * @param id The ID of the user.
```
Comment (souyahia-monk) [inline packages/analytics/src/adapters/adapter.ts:13]: blank line above the `@param` annotation (same for all TSDoc in the PR)

File: packages/analytics/src/adapters/adapter.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+   */
+  setUserProperties: (context: Record<string, Primitive>) => void;
+  /**
+   * Unlink any future events made on that device with that user.
+   */
+  resetUser: () => void;
+  /**
+   * Track a event.
+   * We recommend using a '[object][verb]' format for your event names, where '[object]' is the entity that the behavior relates to, and '[verb]' is the behavior itself.
+   * For example, project created, user signed up, or invite sent.
```
Comment (souyahia-monk) [inline packages/analytics/src/adapters/adapter.ts:29]: Track an event.

We recommend using the...

wrap example between backticks for code formatting like `project created` etc

File: packages/analytics/src/adapters/emptyAdapter.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      console.warn(
+        'Transactions are not supported by the current Monk Analytics Adapter and calling trackEvent will have no effect.',
+      );
+    }
+  }
+
+  setEventsProperties(context?: Record<string, Primitive>): void {
+    if (this.options.showUnsupportedMethodWarnings) {
+      console.warn(
+        'Transactions are not supported by the current Monk Analytics Adapter and calling setEventsProperties will have no effect.',
```
Comment (souyahia-monk) [inline packages/analytics/src/adapters/emptyAdapter.ts:86]: Events are not supported by...

File: packages/analytics/src/react/context.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { createContext } from 'react';
+import { AnalyticsAdapter, EmptyAnalyticsAdapter } from '../adapters';
+
+/**
+ * A React Context that wraps up the analytics methods provided by a Analytics Adapter. This context is initialized by
```
Comment (souyahia-monk) [inline packages/analytics/src/react/context.ts:5]: ...provided by an Analytics Adapter...

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx [hunk 8]
```diff
@@ -103,6 +105,11 @@ export function PhotoCapture({
   const onLastSightTaken = () => {
     startTasks()
       .then(() => {
+        analytics.trackEvent('Inspection completed', {
+          inspectionId,
+          category: 'inspection_completed',
+        });
+        analytics.resetUser();
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx:112]: Why do we reset the user at the end of the inspection?

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx [hunk 9]
```diff
@@ -103,6 +105,11 @@ export function PhotoCapture({
   const onLastSightTaken = () => {
     startTasks()
       .then(() => {
+        analytics.trackEvent('Inspection completed', {
+          inspectionId,
+          category: 'inspection_completed',
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx:110]: What does the `category` tag do exactly?

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCaptureHUD/PhotoCaptureHUD.tsx [hunk 10]
```diff
@@ -96,12 +102,23 @@ export function PhotoCaptureHUD({
   const { t } = useTranslation();
   const [showCloseModal, setShowCloseModal] = useState(false);
   const style = usePhotoCaptureHUDStyle();
+  const { trackEvent } = useAnalytics();
 
   const handleCloseConfirm = () => {
     setShowCloseModal(false);
+    trackEvent('close confirmed', {
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/PhotoCaptureHUD/PhotoCaptureHUD.tsx:109]: Just a quick note : is this the formatting used and recommended by posthog for event names? All lowercase names?

Also, this event should be called `capture closed` imo

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts [hunk 11]
```diff
@@ -174,9 +191,15 @@ export function usePhotoCaptureSightState({
         try {
           const alreadyTakenSights = getSightsTaken(inspectionId, response);
           setSightsTaken(alreadyTakenSights);
-          setSelectedSight(captureSights.filter((s) => !alreadyTakenSights.includes(s))[0]);
+          selectSight(captureSights.filter((s) => !alreadyTakenSights.includes(s))[0]);
           setLastPictureTaken(getLastPictureTaken(inspectionId, response));
           assertInspectionIsValid(inspectionId, response, captureSights, tasksBySight);
+          analytics.trackEvent('inspection started', {
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:197]: event should be called `capture started` imo

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts [hunk 12]
```diff
@@ -174,9 +191,15 @@ export function usePhotoCaptureSightState({
...
+            newInspection: !!alreadyTakenSights,
+            alreadyTakenSights: alreadyTakenSights.length,
+            haveToCaptureSights: captureSights.length,
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:200]: rename to `remainingSights`

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts [hunk 13]
```diff
@@ -162,10 +163,26 @@ export function usePhotoCaptureSightState({
   const [sightsTaken, setSightsTaken] = useState<Sight[]>([]);
   const { getInspection } = useMonkApi(apiConfig);
   const { handleError } = useMonitoring();
+  const analytics = useAnalytics();
+
+  const selectSight = (s: Sight) => {
+    const sightsNotTaken = captureSights.filter((sight) => !sightsTaken.includes(sight)) ?? [];
+    const nextSight = sightsNotTaken.at(sightsNotTaken.indexOf(s) + 1)?.label ?? 'last-sight';
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:170]: instead of `'last-sight'` maybe pass `null`?

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts [hunk 14]
```diff
@@ -198,8 +221,14 @@ export function usePhotoCaptureSightState({
     const updatedSightsTaken = [...sightsTaken, selectedSight];
     setSightsTaken(updatedSightsTaken);
     const nextSight = captureSights.filter((s) => !updatedSightsTaken.includes(s))[0];
+    analytics.trackEvent(`sight captured`, {
+      inspectionId,
+      sight: selectedSight.label,
+      nextSight: nextSight?.label ?? 'last-sight',
```
Comment (souyahia-monk) [inline packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:227]: same as before : use sight ID instead of label and pass `null` if its the last sight

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx (hunk 8 above)
Comment (dlymonk) [reply packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx:113]: Actually, it was for my tests. I will remove it

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx (hunk 9 above)
Comment (dlymonk) [reply packages/inspection-capture-web/src/PhotoCapture/PhotoCapture.tsx:111]: It meant to be used as filter in our dasboards instead of adding all the different events one by one. 
For example, comparing the sum of 3 differents 'category' events versus 4 differents 'category' event.

File: packages/inspection-capture-web/src/PhotoCapture/PhotoCaptureHUD/PhotoCaptureHUD.tsx (hunk 10 above)
Comment (dlymonk) [reply packages/inspection-capture-web/src/PhotoCapture/PhotoCaptureHUD/PhotoCaptureHUD.tsx:109]: The only recommendation from Posthog is to use '[object][verb]' format for event names.

File: documentation/docs/packages/camera-web.md (hunk 1 above)
Comment (dlymonk) [reply documentation/docs/packages/camera-web.md]: Yes it was my auto-correct !

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts (hunk 12 above)
Comment (dlymonk) [reply packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts:200]: Actually, it is the total of sights the user have to take for the whole inspection. `totalSights` is better?

File: packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts (hunk 13 above)
Comment (dlymonk) [reply packages/inspection-capture-web/src/PhotoCapture/hooks/usePhotoCaptureSightState.ts]: i will add an other tag for the label, easier to filter out by part in posthog
